src/java.base/share/classes/java/lang/foreign/GroupLayout.java

```diff
@@ -73,4 +76,136 @@ public sealed interface GroupLayout extends MemoryLayout permits StructLayout, U
      */
     @Override
     GroupLayout withByteAlignment(long byteAlignment);
+
+    /**
+     * {@return a {@link Function} that can project {@linkplain MemorySegment MemorySegments} onto new instances
+     * of the provided {@link Record} {@code type} by means of matching the names of the
+     * record components with the names of the elements in this group layout}
+     * <p>
+     * The mapping between the record type and this memory layout is defined as follows;
+     * <p>
+     * Let <em>R</em> be the provided record {@code type} with the constituent components
+     * <em>C<sub>0</sub></em>, ..., <em>C<sub>N-1</sub></em>, where <em>N</em> is non-negative.
+     * <p>
+     * Let <em>L</em> be this group layout with the elements <em>E<sub>0</sub></em>, ..., <em>E<sub>M-1</sub></em>
+     * , where <em>M</em> {@code >=} <em>N</em>.
+     * <p>
+     * Then, for each <em>C<sub>a</sub></em> (<em>a</em> {@code <} <em>N</em>), there must be a corresponding distinct
```

Comment:
As discussed offline, I think the documentation would be clearer (and more precise) if we:

* Defined some helper function `offsetOf(L, LL)` which gives offset of layout LL in L (no need to specify how it's done, just use it as an explanation device (perhaps hinting at `MemoryLayout::byteOffset`
* Defined a mapping from (offset, layout, class<T>) to a value of type T. This mapping needs to be defined for _all_ layouts
* This allow us to define the sequence case as just using recursion: e.g. given the triple (O, Sequence(E), T[]), for each element E at offset OO = O + offsetOf(Sequence(E), E), we compute the value associated with (OO, E, T). Then we collect all such values into a new T[] array (whose size is the same as that of the sequence layout). I believe this does the right thing even in the case of multi-dimensional arrays.
* Then , recordMapper(G, R) is defined simply as the value associated with the triple (0, G, R).